internal/pkg/manager/nodestatus/nodestatus.go

```diff
@@ -167,12 +169,28 @@ func (r *StatusReconciler) Reconcile(_ context.Context, req reconcile.Request) (
 	// make sure we have all the statuses already
 	hasStatuses := len(nodeStatusList.Items)
 	wantsStatuses := spodDS.Status.DesiredNumberScheduled
-	if wantsStatuses != int32(hasStatuses) {
+	if wantsStatuses > int32(hasStatuses) {
 		logger.Info("Not updating policy: not all statuses are ready",
 			"has", hasStatuses, "wants", wantsStatuses)
 		// Don't reconcile again, let's just wait for another update
 		return reconcile.Result{}, nil
 	}
+	if wantsStatuses < int32(hasStatuses) {
```

Comment:
I think it would be cleaner to do an else-if instead of two ifs
```
if wantsStatuses > int32(hasStatuses) {
 ...
} else if wantsStatuses < int32(hasStatuses) {
 ...
} 

// we know they are equal here
```
this is just to make it clear in the code that we are still continuing the same if